src/managers/slack/slack-commands-parser.js

```diff
@@ -0,0 +1,412 @@
+'use strict';
+
+/**
+ * @module commands-parser
+ * @author Alteh Union (alteh.union@gmail.com)
+ * @license MIT (see the root LICENSE file for details)
+ */
+
+const LangManager = require('../lang-manager');
+
+const BotPublicError = require('../../utils/bot-public-error');
+const SlackUtils = require('../../utils/slack-utils');
+
+const BotTable = require('../../mongo_classes/bot-table');
+const ServerSettingsTable = require('../../mongo_classes/server-settings-table');
+const UserSettingsTable = require('../../mongo_classes/user-settings-table');
+
+const SlackCommand = require('../../commands/commands_slack/slack-command');
+
+const HelpCommand = require('../../commands/commands_slack/help-command');
+const PingCommand = require('../../commands/commands_slack/ping-command');
+
+
+/**
+ * The defined Slack command classes.
+ * @type {Array<constructor>}
+ */
+const SlackCommands = Object.freeze([
+  PingCommand
+]);
+
+/**
+ * The defined private Slack command classes.
+ * @type {Array<constructor>}
+ */
+const SlackPrivateCommands = Object.freeze([]);
+
+/**
+ * Parses and launches execution of the Bot's commands.
+ * @alias CommandsParser
+ */
+class SlackCommandsParser {
+  /**
+   * Constructs an instance of the class.
+   * @param {Context} context the Bot's context
+   * @param {WebClient} webClient for the Slack API
+   */
+  constructor(context, webClient) {
```

Comment:
Can we make the Slack web client a member of the context? I'm afraid we will need it in quite some places while working with advanced Slack commands, so having it inside the context would help to reduce the count of arguments.